docs/question/html.md

```diff
@@ -1,13 +1,37 @@
-# Using Markdown in questions
+# `question.html`
 
-HTML and custom elements are great for flexibility and expressiveness. However, they're not great for working with large amounts of text, formatting text, and so on. [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) is a lightweight plaintext markup syntax that's ideal for authoring simple but rich text. To enable this, PrairieLearn adds a special `<markdown>` tag to questions. When a `<markdown>` block is encountered, its contents are converted to HTML. Here's an example `question.html` that utilizes this element:
+## Template file
+
+All `question.html` files are processed with the [Mustache template engine](https://mustache.github.io/mustache.5.html), which allows for basic dynamic rendering of HTML. For example, you can use the following syntax to conditionally render a piece of HTML:
+
+```html title="question.html"
+{{#params.show}}
+<p>This will only show up if params.show is true</p>
+{{/params.show}}
+```
+
+and this syntax to render over an array of strings (the `.` represents the current item in the array):
+
+```html title="question.html"
+{{#params.items}}
+<p>{{.}}</p>
+{{/params.items}}
+```
+
+!!! tip
+
+    If you use triple-braces (e.g. `{{{params.html}}}`) then raw HTML is substituted (don't use this unless you know you need it).
+
+## Markdown in questions
+
+HTML and custom elements are great for flexibility and expressiveness. However, they're not great for working with large amounts of text, formatting text, and so on. [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) is a lightweight plaintext markup syntax that's ideal for authoring simple but rich text.
+
+You can use the special `<markdown>` tag to automatically convert its contents to HTML. Here's an example `question.html` that utilizes this element:
 
 <!-- prettier-ignore -->
 ```html title="question.html"
 <markdown>
-# Hello, world!
-
-This is some **Markdown** text.
+# Hello, world! This is some **Markdown** text.
```

Comment:
I think this was not intended to be combined...

Reply:
My lines `8-10` look like:

````
```html title="question.html"
{{#params.show}}
<p>This will only show up if params.show is true</p>
{{/params.show}}
```
````

Reply:
Did you comment on the wrong thread?

Reply:
Oops, I didn't understand what you meant. Fixed in ba0c214210eee08d5d93aec6e2de7d1ed3537069.